Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java

```diff
@@ -102,18 +115,47 @@ public String getEstimatedBalance() {
      * least 1 confirmation and pending transactions created by our own wallet which have been propagated across
      * the network. Whether we <i>actually</i> have the private keys or not is irrelevant for this balance type.
      */
-    public String getAvailableBalance() {
-        return kit.wallet().getBalance(Wallet.BalanceType.AVAILABLE).toFriendlyString();
+    public String getAvailableBalance() throws WalletNotInitialized {
+        return getWallet().getBalance(Wallet.BalanceType.AVAILABLE).toFriendlyString();
     }
 
-    public void cleanUp() {
-        fileWallet.delete();
-        fileSpvBlockchain.delete();
+    public void restoreFromSeed(List<String> mnemonicCode) {
+        DeterministicSeed seed = new DeterministicSeed(mnemonicCode, null, "", 1539388800);
+        // Shut down synchronization and restart it with the new seed.
+        Runnable setupWalletFromBackup = () -> {
+            setupWalletKit(seed);
+            try {
+                getKit().startAsync();
+            } catch (WalletNotInitialized walletNotInitialized) {
+                walletNotInitialized.printStackTrace();
+            }
+        };
+        try {
+            getKit().addListener(new Service.Listener() {
```

Comment:
could you transform it to lambda expression?

Reply:
Unfortunately, I can not, because the Service .Listener is an abstract class, and not interface :/

Reply:
Unfortunately, I can not, because the Service .Listener is an abstract class, and not interface :/